Lib/test/_test_multiprocessing.py

```diff
@@ -766,6 +766,33 @@ def __reduce__(self):
             q.put(True)
             self.assertTrue(q.get(timeout=0.1))
 
+    def test_queue_feeder_on_queue_feeder_error(self):
+        # bpo-30006: verify feeder handles exceptions using the
+        # _on_queue_feeder_error hook.
+        if self.TYPE != 'processes':
+            self.skipTest('test not appropriate for {}'.format(self.TYPE))
+
+        class NotSerializable(object):
+            def __init__(self):
+                self.pass_test = True
```

Comment:
Ok then to make the test more readable we should have two flags:

- one named `reduce_was_called` set to `False` in `__init__` and subsequently set to True in `__reduce__`
- another named `on_queue_feeder_error_was_called` set to `False` in `__init__` and subsequently set to True in `_on_queue_feeder_error` by the Queue itself if `e` and `obj` have the expected type.

And the reverse the order of  ` q.put(unserializable_obj)` and `q.put(True)` to as to make sure that we can make `self.assertTrue` on the flags after `self.assertTrue(q.get(timeout=0.1))` has succeeded.